tests/unit/igraph_coloring.c

```diff
@@ -0,0 +1,56 @@
+#include <igraph.h>
+
+#include "test_utilities.h"
+
+igraph_error_t verify_coloring(igraph_t *graph, igraph_vector_int_t *colors){
+    /* Verify that the colouring is valid, i.e. no two adjacent vertices have the same colour. */
+    igraph_integer_t i;
+    /* Store the edge count to avoid the overhead from igraph_ecount in the for loop. */
+    igraph_integer_t no_of_edges = igraph_ecount(graph);
+    for (i = 0; i < no_of_edges; ++i) {
+        if ( VECTOR(*colors)[ IGRAPH_FROM(graph, i) ] == VECTOR(*colors)[ IGRAPH_TO(graph, i) ]  ) {
+            printf("Inconsistent coloring! Vertices %" IGRAPH_PRId " and %" IGRAPH_PRId " are adjacent but have the same color.\n",
+                   IGRAPH_FROM(graph, i), IGRAPH_TO(graph, i));
+            return IGRAPH_FAILURE;
```

Comment:
Instead of returning an error code, make this function `void` and replace the `printf` with the `IGRAPH_FATALF` macro. The argument are exactly the same as that of `printf`, so they don't need to be modified.

Normally, we use `IGRAPH_ASSERT` in tests, which calls `igraph_fatal()` internally. Here `IGRAPH_FATALF` si a better fit as it can print a more informative failure message.

These functions also ensure that a stack trace will be printed on failure, so one can tell which exact `verify_coloring()` call failed, if there are multiple.